### PR TITLE
Add symbol key tests

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi What were these tests needed for?

**Comment by CyrusNajmabadi** [comment]
I was investigating a prism report and i thought there might be an issue with these unnamed symbols.  I looked and saw no tests so i added tests to see if they woudl cause the crash.  But they did not, showing that these scenarios work.  I wanted to at least check the tests in to help ensure that stays the case (and also doesn't regress in the future) :)

File: src/Workspaces/CoreTest/SymbolKeyTests.cs [hunk 1]
```diff
@@ -113,6 +113,62 @@ public class B { };
...
+            var source = @"
+
+public class C
+{
+    event System.Action a,;
+}
+";
+            var compilation = GetCompilation(source, LanguageNames.CSharp);
+            TestRoundTrip(GetDeclaredSymbols(compilation), compilation);
+        }
```
Comment (genlu) [inline src/Workspaces/CoreTest/SymbolKeyTests.cs:170]: Based on the test names, it looks like `GetDeclaredSymbols(compilation)` doesn't return the symbol from erroneous declaration? Is that correct? Maybe consider assert on returned symbols like the test below to make it clear?

**Review by genlu — APPROVED** [review]

File: src/Workspaces/CoreTest/SymbolKeyTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CoreTest/SymbolKeyTests.cs:170]: ah.  it does actually return them.  but an asseertion makes sense.  wil. add.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.cs [hunk 2]
```diff
@@ -188,6 +194,8 @@ public static bool CanCreate(ISymbol symbol, CancellationToken cancellationToken
                 return default;
             }
 
+            var language = reader.ReadString();
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.cs:197]:
```suggestion
            // Read out the language info which was included just for diagnostic purposes.
            var language = reader.ReadString();
```

**Pull request merged**